lib/ex_rlp/decode_item.ex

```diff
@@ -0,0 +1,1778 @@
+    <<item::binary-size(item_length), new_tail::binary>> = tail
+    do_decode_item(new_tail, item)
+  end
+
+  defp do_decode_item(<<156, tail::binary>>, result) do
+    item_length = 156 - 128
+    <<item::binary-size(item_length), new_tail::binary>> = tail
+    do_decode_item(new_tail, [item | result])
+  end
+
+  defp do_decode_item(<<157, tail::binary>>, nil) do
+    item_length = 157 - 128
+    <<item::binary-size(item_length), new_tail::binary>> = tail
+    do_decode_item(new_tail, item)
+  end
+
+  defp do_decode_item(<<157, tail::binary>>, result) do
+    item_length = 157 - 128
+    <<item::binary-size(item_length), new_tail::binary>> = tail
+    do_decode_item(new_tail, [item | result])
+  end
+
+  defp do_decode_item(<<158, tail::binary>>, nil) do
+    item_length = 158 - 128
+    <<item::binary-size(item_length), new_tail::binary>> = tail
+    do_decode_item(new_tail, item)
+  end
+
+  defp do_decode_item(<<158, tail::binary>>, result) do
+    item_length = 158 - 128
+    <<item::binary-size(item_length), new_tail::binary>> = tail
+    do_decode_item(new_tail, [item | result])
+  end
+
+  defp do_decode_item(<<159, tail::binary>>, nil) do
+    item_length = 159 - 128
+    <<item::binary-size(item_length), new_tail::binary>> = tail
+    do_decode_item(new_tail, item)
+  end
+
+  defp do_decode_item(<<159, tail::binary>>, result) do
+    item_length = 159 - 128
+    <<item::binary-size(item_length), new_tail::binary>> = tail
+    do_decode_item(new_tail, [item | result])
+  end
+
+  defp do_decode_item(<<160, tail::binary>>, nil) do
+    item_length = 160 - 128
+    <<item::binary-size(item_length), new_tail::binary>> = tail
+    do_decode_item(new_tail, item)
+  end
+
+  defp do_decode_item(<<160, tail::binary>>, result) do
+    item_length = 160 - 128
+    <<item::binary-size(item_length), new_tail::binary>> = tail
+    do_decode_item(new_tail, [item | result])
+  end
+
+  defp do_decode_item(<<161, tail::binary>>, nil) do
+    item_length = 161 - 128
+    <<item::binary-size(item_length), new_tail::binary>> = tail
+    do_decode_item(new_tail, item)
+  end
+
+  defp do_decode_item(<<161, tail::binary>>, result) do
+    item_length = 161 - 128
+    <<item::binary-size(item_length), new_tail::binary>> = tail
+    do_decode_item(new_tail, [item | result])
+  end
+
+  defp do_decode_item(<<162, tail::binary>>, nil) do
+    item_length = 162 - 128
+    <<item::binary-size(item_length), new_tail::binary>> = tail
+    do_decode_item(new_tail, item)
+  end
+
+  defp do_decode_item(<<162, tail::binary>>, result) do
+    item_length = 162 - 128
+    <<item::binary-size(item_length), new_tail::binary>> = tail
+    do_decode_item(new_tail, [item | result])
+  end
+
+  defp do_decode_item(<<163, tail::binary>>, nil) do
+    item_length = 163 - 128
+    <<item::binary-size(item_length), new_tail::binary>> = tail
+    do_decode_item(new_tail, item)
+  end
+
+  defp do_decode_item(<<163, tail::binary>>, result) do
+    item_length = 163 - 128
```

Comment:
wouldn't it be faster if these values were precomputed?

Reply:
yes!

Reply:
I'll add that.